pymongoexplain/__main__.py

```diff
@@ -13,48 +13,35 @@
 # limitations under the License.
 
 
-from pymongo import monitoring, MongoClient
 from pymongo.collection import Collection
+from .explainable_collection import ExplainCollection
+
 import sys
-from bson.son import SON
+import logging
 
-class CommandLogger(monitoring.CommandListener):
-    def __init__(self):
-        self.payloads = []
+FORMAT = '%(asctime)s %(levelname)s %(module)s %(message)s'
+logging.basicConfig(format=FORMAT, level=logging.INFO)
 
-    def started(self, event):
-        print("started")
-        self.payloads.append(event.command)
+old_function_names = ["update_one", "replace_one", "update_many", "delete_one",
+                      "delete_many", "aggregate", "watch", "find", "find_one",
+                      "find_one_and_delete", "find_one_and_replace",
+                      "find_one_and_update", "count_documents",
+                      "estimated_document_count", "distinct"]
+old_functions = [getattr(Collection, i) for i in old_function_names]
 
-    def succeeded(self, event):
-        pass
 
-    def failed(self, event):
-        pass
+def make_func(old_func, old_func_name):
+    def new_func(self: Collection, *args, **kwargs):
+        res = getattr(ExplainCollection(self),old_func_name)(*args, **kwargs)
+        logging.info("%s explain response: %s", old_func_name, res)
+        return old_func(self, *args, **kwargs)
+    return new_func
 
-if __name__ == '__main__':
 
+for old_func, old_func_name in zip(old_functions, old_function_names):
+    setattr(Collection, old_func_name, make_func(old_func, old_func_name))
```

Comment:
Did you confirm that this patches `Collection` no matter what namespace it is imported from? In other words, do we know that the following two cases would both see the same behavior:

```
import pymongo

pymongo.Collection(...).find(...)
```

and 

```
from pymongo import Collection

Collection(...).find(...)
```

Reply:
We also want the patching to work when PyMongo itself returns a Collection (e.g. via `Database.get_collection`). Checking once manually on your local machine should suffice for now, though maybe you can add a note about testing these cases to the ticket Shane has asked you to open above.

Reply:
I tested those two cases and it still works.